tests/unit/misc/test_cmdhistory.py

```diff
@@ -0,0 +1,196 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Alexander Cogneau (acogneau) <alexander.cogneau@gmail.com>:
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for misc.History."""
+
+import pytest
+
+from qutebrowser.misc.cmdhistory import (History, HistoryEmptyError,
+                                         HistoryEndReachedError)
+
+
+HISTORY = ['first', 'second', 'third', 'fourth', 'fifth']
+
+CONFIG_NOT_PRIVATE = {'general': {'private-browsing': False}}
+CONFIG_PRIVATE = {'general': {'private-browsing': True}}
+
+
+class TestConstructor:
+
+    """Tests for the constructor."""
+
+    def test_no_history(self):
+        hist = History()
+        # .history should equal []
+        assert len(hist.history) == 0
+
+    def test_history(self):
+        hist = History(history=HISTORY)
+        assert hist.history == HISTORY
+
+
+class TestCommandHistory:
+
+    """Tests for Command History."""
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.hist = History(history=HISTORY)
+
+    def test_is_browsing(self):
+        """Test is_browsing()."""
+
+        self.hist._tmphist = None
+        assert not self.hist.is_browsing()
+
+        self.hist._tmphist = HISTORY
+        assert self.hist.is_browsing()
```

Comment:
I think this should be a parametrized test, so it's split into two tests and it's easier to add new testcases in the future:

```python
@pytest.mark.parametrize(`tmphist, expected', [(None, False), (HISTORY, True)])
def test_is_browsing(hist, tmphist, expected):
    hist._tmphist = tmphist
    assert hist.is_browsing() == expected
```

(the example is with all the above comments applied, but untested)